Support custom pipeline background image styling

analytically opened the pull request:
Add support for configurable filter setting for pipeline background images. This enhances the visual customization options for pipelines by allowing users to set:

- Custom filter effects (defaults to grayscale(100%) opacity(30%) as before)

# Release Note

* Add `background_filter` option for pipeline background images which takes in string of [CSS filters](https://developer.mozilla.org/en-US/docs/Web/CSS/filter). Defaults to the current filters `opacity(30%) grayscale(100%)`

taylorsilva commented:
Just a note-to-self, we should document these: https://github.com/concourse/docs/blob/70e8fd6ada55064074bc0d98666ca0219fa8d637/lit/docs/pipelines.lit#L109-L135

taylorsilva reviewed (changes requested):
Very nice! I think we can actually remove the `background_opacity` field because `background_filter` can be used to combine multiple filters like `opacity()`: https://developer.mozilla.org/en-US/docs/Web/CSS/filter

I made this filter:
```yaml
background_filter: blur(5px)grayscale(100%)
```

and got:

<img width="1776" alt="image" src="https://github.com/user-attachments/assets/0842c748-494f-4c7b-88c7-5529c399cb78" />

Adding `opacity()`
```yaml
background_filter: blur(5px)grayscale(100%)opacity(50%)
```
<img width="1757" alt="image" src="https://github.com/user-attachments/assets/f47398b5-7a25-4979-8dd7-dc6c6a9664ee" />

`background_filter` is all we really need :)

analytically commented:
Currently opacity is set to 30%. If we'd just allow the filter to be set, what do we do with the default value? Not set it? People might expect it to be kept at 30% when just setting the filter?

analytically commented:
Having both opacity and filter is more flexible IMO. You can just set the image and opacity without thinking about filter. Or just image and filter, then an opacity of 30% is applied by default.

analytically commented:
Or, we remove the default for opacity into the filter.

analytically committed:
Support custom pipeline background image styling

Add support for configurable filter setting for pipeline background images. This enhances the visual customization options for pipelines by allowing users to set:

- Custom filter effects (defaults to grayscale(100%) opacity(30%) as before)

Signed-off-by: Mathias Bogaert <mathias.bogaert@gmail.com>
